inspirehep/modules/warnings/ext.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Extension that suppresses warnings.
+
+By default Invenio-Base enables deprecation warnings. This extension
+disables them, unless ``DEBUG_SHOW_DEPRECATION_WARNINGS`` is ``True``.
+
+This extension also disables SQLAlchemy warnings, unless the variable
+``DEBUG_SHOW_SQLALCHEMY_WARNINGS` is set to ``True``.
```

Comment:
@jacquerie they introduced in Invenio 2 deprecation warnings but given the need for reimplementing everything from scratch in Invenio 3, and given it's bleeding edge status (although the only code really runnable on production), and our pressure towards reaching quickly Invenio 3 they decided not to use deprecation warnings (which are slowing down development) until they make the stable release 1.0. I think it was a necessary evil. We can't ask for perfect situation and speed all in one step. Especially since INSPIRE itself was originally behind the request to have the hybrid of Invenio 2 to allow legacy and Flask to coexist which ended up slowing down reimplementation of Invenio. 

After 1.0 API will be declared stable and will evolve using deprecation warning techniques.

Also if they break the API and fix them in one day I think it's quick enough. Of course we have to be careful that if we deploy on production we must use pinned versions that we know they work.

We could use for help us this: https://github.com/inveniosoftware/requirements-builder

Regarding this PR, if indeed these Warnings are the only to go LGTM.